base/array.jl

```
@@ -237,16 +237,29 @@ function _collect(cont, itr, ::HasEltype, isz::SizeUnknown)
     return a
 end
 
+if isdefined(:Base)
+@eval function _default_eltype(itrt::ANY)
+    $(Expr(:meta, :pure))
+    try
+        rt = return_types(first, (itrt,))
+        if length(rt) == 1 && isleaftype(rt[1])
+            return rt[1]
+        end
+    end
```

probably not a good idea to ignore all errors